library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java

```diff
@@ -0,0 +1,223 @@
+package com.novoda.downloadmanager.lib;
+
+import android.support.annotation.IntDef;
+import android.support.annotation.NonNull;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class used to query the batches table. Use {@link BatchQuery.Builder} in order to build it.
+ */
+public class BatchQuery {
+    public static final BatchQuery ALL = new BatchQuery(null, null, null);
+    private static final int LOW_END_FAILED_STATUS_CODE = 400;
+    private static final int HIGH_END_FAILED_STATUS_CODE = 600;
+
+    private final String selection;
+    private final String sortOrder;
+    private final String[] selectionArguments;
+
+    BatchQuery(String selection, String[] selectionArguments, String sortOrder) {
+        this.selection = selection;
+        this.sortOrder = sortOrder;
+        this.selectionArguments = selectionArguments;
+    }
+
+    String getSelection() {
+        return selection;
+    }
+
+    String getSortOrder() {
+        return sortOrder;
+    }
+
+    String[] getSelectionArguments() {
+        return selectionArguments;
+    }
+
+    public static class Builder {
+
+        private final Criteria.Builder builder;
+
+        private static final String ORDER_BY_LIVENESS = String.format("CASE batch_status "
+                        + "WHEN %1$d THEN 1 "
+                        + "WHEN %2$d THEN 2 "
+                        + "WHEN %3$d THEN 3 "
+                        + "WHEN %4$d THEN 4 "
+                        + "WHEN %5$d THEN 5 "
+                        + "ELSE 2 "
+                        + "END",
+                Downloads.Impl.STATUS_RUNNING,
+                Downloads.Impl.STATUS_PENDING,
+                Downloads.Impl.STATUS_PAUSED_BY_APP,
+                Downloads.Impl.STATUS_BATCH_FAILED,
+                Downloads.Impl.STATUS_SUCCESS
+        );
+
+        private Criteria.Builder criteriaIdBuilder;
+        private Criteria.Builder criteriaStatusBuilder;
+
+        public Builder() {
+            builder = new Criteria.Builder();
+        }
+
+        /**
+         * Filter by batch id
+         *
+         * @param id id of the batch
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withId(long id) {
```

Comment:
The below `criteriaIdBuilder` is created every time in order to reset the id to the last one if multiple subsequent calls to `withId()` are made.